infra/compiler.md

```diff
@@ -234,12 +234,19 @@ Conditional expressions are also supported:
     >>> Test.expr("x if y else z")
     (y ? x : z)
 
-One weird case is list comprehensions. We compile those to `map` and `filter`:
-
+One weird case is list comprehensions. We compile those to `map` and
+`asyncfilter`, which can get a bit ugly:
+
+    >>> Test.expr("[x for x in v]")
+    v
```

Comment:
aside: did you special-case no-ops somewhere?

Reply:
Not really. Before all list comprehensions had a “map” component but not necessarily a “filter” component. So I didn’t output “filter” code if it wasn’t there, but I always output “map”. Now there are some pure-filter list comprehensions, plus the “map” code got uglier to deal with async, so I made it not output a “map” unless it does something. Put it together, and no-ops now do nothing.